### PR TITLE
Document decision to let pipelines proceed

### PR DESCRIPTION
**Pull request by ralphbean**

### CODE REVIEW

File: ADR/0014-let-pipelines-proceed.md
Comment (arewm) [inline ADR/0014-let-pipelines-proceed.md:41]: What is considered a `lower Environment` here?

File: ADR/0014-let-pipelines-proceed.md
Comment (arewm) [inline ADR/0014-let-pipelines-proceed.md:36]: Would it be up to both the UI and PaC to render the output to users? I would expect that at least some GitHub users/organizations would want to enforce checks to pass before merging. How might this be able to be accomplished?

Instead of just having the UI/PaC go through all TaskRuns, would it be reasonable to have a "finalizer" Task which can aggregate and summarize the result of the previous scanning/linting TaskRuns. 

I don't _really_ like stating a further reliance on `HACBS_TEST_OUTPUT` as I think that we should try to come to some other generic pattern which doesn't propagate a tie-down to an old product name.

File: ADR/0014-let-pipelines-proceed.md
Comment (arewm) [inline ADR/0014-let-pipelines-proceed.md:33]: Are there any types of Tasks which should explicitly _not_ always succeed?

**Review by arewm — COMMENTED** [review]
I see the footnote about this being a verbal decision that has been made already. If this ADR should be approved as-is with that in mind, then I can just leave my comments as unresolved as needed, coming back around to them at some point in time later.

File: ADR/0014-let-pipelines-proceed.md
Comment (ralphbean) [reply ADR/0014-let-pipelines-proceed.md:41]: Left intentionally vague for the purposes of the ADR. Anything that is "not production". Here, we're still assuming that the user uses the managed-workspace release model via the release service.

File: ADR/0014-let-pipelines-proceed.md
Comment (ralphbean) [reply ADR/0014-let-pipelines-proceed.md:36]: > I don't really like stating a further reliance on HACBS_TEST_OUTPUT as I think that we should try to come to some other generic pattern which doesn't propagate a tie-down to an old product name.

Agreed. It is one of the reasons we've been holding off on moving forwards with [STONE-459](https://issues.redhat.com/browse/STONE-459), but it's getting too late now and I don't want to wait for [TKNECO-12](https://issues.redhat.com/browse/TKNECO-12). Once we get a recommendation there, we should land new decisions overturning this ADR and ADR 13.

> Would it be up to both the UI and PaC to render the output to users? I would expect that at least some GitHub users/organizations would want to enforce checks to pass before merging. How might this be able to be accomplished?

Yeah, that's the idea in STONE-459. PaC wouldn't really render the output into a UI element - it would render the output into a call to the GitHub Checks API, which would make a successful TaskRun with "errors" in its HACBS_TEST_OUTPUT end up being represented as a "failing" check in the GitHub UI.

> Instead of just having the UI/PaC go through all TaskRuns, would it be reasonable to have a "finalizer" Task which can aggregate and summarize the result of the previous scanning/linting TaskRuns.

This is possible, but PaC and the UI both already need to deal with per-task stuff anyways. PaC submits a check result for each task (in GitHub you can see a table of all ~dozen steps performed in a pipelinerun) and in the UI we render the whole pipeline run with the ability to click through into the logs of a task. Having both of those systems be able to render errors on a per-task level directs the user to exactly where they need to click to go see the stdout from scanner $foo.

File: ADR/0014-let-pipelines-proceed.md
Comment (ralphbean) [reply ADR/0014-let-pipelines-proceed.md:33]: Not clear to me at this time.

The git clone task should fail if it fails (because it can't provide output to the next stages of the pipelinerun). The buildah task should fail if it fails (because it can't produce the image... which is needed by subsequent stages of the pipelinerun?)

**Comment by ralphbean** [comment]
> I see the footnote about this being a verbal decision that has been made already. If this ADR should be approved as-is with that in mind, then I can just leave my comments as unresolved as needed, coming back around to them at some point in time later.

I think the substance should be approved. I'm happy to integrate feedback on anything that's unclear or unrepresentative of the existing direction. If we disagree about the direction forwards, that's a good case to followup with an ADR that overrides this one - documenting a new point-in-time decision.

**Review by sbose78 — APPROVED** [review]

**Comment by sbose78** [comment]
Seems fair to me.

**Review by hongweiliu17 — APPROVED** [review]
Just a few minor comment. LGTM.

File: ADR/0013-integration-service-api-contracts.md
Comment (ralphbean) [reply ADR/0013-integration-service-api-contracts.md:9]: Yes, it should get rendered as html at https://redhat-appstudio.github.io/book/ADR/ after merge.

**Review by arewm — APPROVED** [review]

File: ADR/0013-integration-service-api-contracts.md
Comment (hongweiliu17) [reply ADR/0013-integration-service-api-contracts.md:9]: I see now. Thank you for the link!

File: ADR/0014-let-pipelines-proceed.md
Comment (hongweiliu17) [inline ADR/0014-let-pipelines-proceed.md:51]:
```suggestion
[Environments]: ../ref/application-environment-api.html#environment
```

**Pull request merged**
